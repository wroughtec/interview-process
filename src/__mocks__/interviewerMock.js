export const mockData = {
  results: [
    {
      name: { first: 'firstName0', last: 'lastName0' },
      location: {
        city: 'city'
      },
      login: {
        uuid: '0a985e0b-9383-43d7-bcf3-f8b8f597451e0'
      },
      picture: {
        thumbnail: 'thumbnail'
      }
    },
    {
      name: { first: 'firstName1', last: 'lastName1' },
      location: {
        city: 'city1'
      },
      login: {
        uuid: '0a985e0b-9383-43d7-bcf3-f8b8f597451e1'
      },
      picture: {
        thumbnail: 'thumbnail1'
      }
    },
    {
      name: { first: 'firstName2', last: 'lastName2' },
      location: {
        city: 'city2'
      },
      login: {
        uuid: '0a985e0b-9383-43d7-bcf3-f8b8f597451e2'
      },
      picture: {
        thumbnail: 'thumbnail2'
      }
    },
    {
      name: { first: 'firstName3', last: 'lastName3' },
      location: {
        city: 'city3'
      },
      login: {
        uuid: '0a985e0b-9383-43d7-bcf3-f8b8f597451e3'
      },
      picture: {
        thumbnail: 'thumbnail3'
      }
    },
    {
      name: { first: 'firstName4', last: 'lastName4' },
      location: {
        city: 'city4'
      },
      login: {
        uuid: '0a985e0b-9383-43d7-bcf3-f8b8f597451e4'
      },
      picture: {
        thumbnail: 'thumbnail4'
      }
    }
  ]
};
