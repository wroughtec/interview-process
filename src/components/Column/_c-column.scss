.c-column {
  display: flex;
  margin-bottom: 1rem;

  flex-direction: column;
}

.c-column__header {
  text-align: center;
}

.c-column__body {
  height: 100%;
  padding: .5rem;

  background-color: var(--columnBackground);
}

.c-column__title {
  margin-top: 0;
  margin-bottom: 1.5rem;

  font-size: calc(1.25rem + .15vw);
}
