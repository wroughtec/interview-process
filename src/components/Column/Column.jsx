// @flow
import React, { Component } from 'react';
import { Card } from 'components/Card/Card';
import { cityFilter } from 'consts/filters';
import './_c-column.scss';

type Props = {
  title: string,
  data: any[],
  currentPosition: string,
  filter: string,
  filterBy: string,
  btnRight: string,
  btnLeft: string,
  handleBtnClick: Function
};

export class Column extends Component<Props> {
  filterCards = data => {
    const { filter, filterBy } = this.props;
    switch (filterBy) {
      case cityFilter:
        return data.filter(item => {
          const {
            location: { city }
          } = item;

          return city.substring(0, filter.length).toLowerCase() === filter;
        });

      default:
        return data.filter(item => {
          const {
              name: { first, last }
            } = item,
            fullName = `${first} ${last}`;

          return fullName.substring(0, filter.length).toLowerCase() === filter;
        });
    }
  };

  displayCards = () => {
    const { data, btnLeft, btnRight, handleBtnClick, currentPosition, filter } = this.props;

    let cards = null,
      filteredData = data;

    if (data && data.length && filter) {
      filteredData = this.filterCards(data);
    }

    if (filteredData && filteredData.length) {
      cards = filteredData.map(card => {
        const {
            name: { first, last },
            picture: { thumbnail },
            location: { city },
            login: { uuid }
          } = card,
          fullName = `${first} ${last}`;

        return (
          <Card
            name={fullName}
            img={thumbnail}
            location={city}
            key={uuid}
            id={uuid}
            btnLeft={btnLeft}
            btnRight={btnRight}
            handleBtnClick={handleBtnClick}
            currentPosition={currentPosition}
          />
        );
      });
    }

    return cards;
  };

  render() {
    const { title } = this.props,
      cards = this.displayCards();
    return (
      <div className="c-column">
        <header className="c-column__header">
          <h2 className="c-column__title">{title}</h2>
        </header>
        <section className="c-column__body">{cards}</section>
      </div>
    );
  }
}
