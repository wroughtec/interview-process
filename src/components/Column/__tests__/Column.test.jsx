import React from 'react';
import { shallow } from 'enzyme';
import { Column } from 'components/Column/Column';
import { mockData } from 'mocks/interviewerMock';

describe('Column', () => {
  it('should render correctly with', () => {
    const props = {
      title: 'title',
      data: [mockData.results[0]],
      currentPosition: 'currentPosition',
      filter: '',
      filterBy: '',
      btnRight: 'btnRight',
      btnLeft: 'btnLeft',
      handleBtnClick: Function
    };
    const component = shallow(<Column {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should render correctly with city filter', () => {
    const props = {
      title: 'title',
      data: [mockData.results[0]],
      currentPosition: 'currentPosition',
      filter: 'city',
      filterBy: 'city',
      btnRight: 'btnRight',
      btnLeft: 'btnLeft',
      handleBtnClick: Function
    };
    const component = shallow(<Column {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should render correctly with name filter', () => {
    const props = {
      title: 'title',
      data: [mockData.results[0]],
      currentPosition: 'currentPosition',
      filter: 'first',
      filterBy: 'name',
      btnRight: 'btnRight',
      btnLeft: 'btnLeft',
      handleBtnClick: Function
    };
    const component = shallow(<Column {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('no data render', () => {
    const props = {
      title: 'title',
      data: [],
      currentPosition: 'currentPosition',
      filter: 'first',
      filterBy: 'name',
      btnRight: 'btnRight',
      btnLeft: 'btnLeft',
      handleBtnClick: Function
    };
    const component = shallow(<Column {...props} />);
    expect(component).toMatchSnapshot();
  });
});
