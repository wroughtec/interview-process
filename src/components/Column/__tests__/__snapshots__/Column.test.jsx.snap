// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column no data render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Column
    btnLeft="btnLeft"
    btnRight="btnRight"
    currentPosition="currentPosition"
    data={Array []}
    filter="first"
    filterBy="name"
    handleBtnClick={[Function]}
    title="title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="c-column__header"
        >
          <h2
            className="c-column__title"
          >
            title
          </h2>
        </header>,
        <section
          className="c-column__body"
        />,
      ],
      "className": "c-column",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h2
            className="c-column__title"
          >
            title
          </h2>,
          "className": "c-column__header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "title",
            "className": "c-column__title",
          },
          "ref": null,
          "rendered": "title",
          "type": "h2",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": null,
          "className": "c-column__body",
        },
        "ref": null,
        "rendered": null,
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="c-column__header"
          >
            <h2
              className="c-column__title"
            >
              title
            </h2>
          </header>,
          <section
            className="c-column__body"
          />,
        ],
        "className": "c-column",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h2
              className="c-column__title"
            >
              title
            </h2>,
            "className": "c-column__header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "title",
              "className": "c-column__title",
            },
            "ref": null,
            "rendered": "title",
            "type": "h2",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": null,
            "className": "c-column__body",
          },
          "ref": null,
          "rendered": null,
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Column should render correctly with 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Column
    btnLeft="btnLeft"
    btnRight="btnRight"
    currentPosition="currentPosition"
    data={
      Array [
        Object {
          "location": Object {
            "city": "city",
          },
          "login": Object {
            "uuid": "0a985e0b-9383-43d7-bcf3-f8b8f597451e0",
          },
          "name": Object {
            "first": "firstName0",
            "last": "lastName0",
          },
          "picture": Object {
            "thumbnail": "thumbnail",
          },
        },
      ]
    }
    filter=""
    filterBy=""
    handleBtnClick={[Function]}
    title="title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="c-column__header"
        >
          <h2
            className="c-column__title"
          >
            title
          </h2>
        </header>,
        <section
          className="c-column__body"
        >
          <Card
            btnLeft="btnLeft"
            btnRight="btnRight"
            currentPosition="currentPosition"
            handleBtnClick={[Function]}
            id="0a985e0b-9383-43d7-bcf3-f8b8f597451e0"
            img="thumbnail"
            location="city"
            name="firstName0 lastName0"
          />
        </section>,
      ],
      "className": "c-column",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h2
            className="c-column__title"
          >
            title
          </h2>,
          "className": "c-column__header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "title",
            "className": "c-column__title",
          },
          "ref": null,
          "rendered": "title",
          "type": "h2",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Card
              btnLeft="btnLeft"
              btnRight="btnRight"
              currentPosition="currentPosition"
              handleBtnClick={[Function]}
              id="0a985e0b-9383-43d7-bcf3-f8b8f597451e0"
              img="thumbnail"
              location="city"
              name="firstName0 lastName0"
            />,
          ],
          "className": "c-column__body",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0a985e0b-9383-43d7-bcf3-f8b8f597451e0",
            "nodeType": "class",
            "props": Object {
              "btnLeft": "btnLeft",
              "btnRight": "btnRight",
              "currentPosition": "currentPosition",
              "handleBtnClick": [Function],
              "id": "0a985e0b-9383-43d7-bcf3-f8b8f597451e0",
              "img": "thumbnail",
              "location": "city",
              "name": "firstName0 lastName0",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="c-column__header"
          >
            <h2
              className="c-column__title"
            >
              title
            </h2>
          </header>,
          <section
            className="c-column__body"
          >
            <Card
              btnLeft="btnLeft"
              btnRight="btnRight"
              currentPosition="currentPosition"
              handleBtnClick={[Function]}
              id="0a985e0b-9383-43d7-bcf3-f8b8f597451e0"
              img="thumbnail"
              location="city"
              name="firstName0 lastName0"
            />
          </section>,
        ],
        "className": "c-column",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h2
              className="c-column__title"
            >
              title
            </h2>,
            "className": "c-column__header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "title",
              "className": "c-column__title",
            },
            "ref": null,
            "rendered": "title",
            "type": "h2",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Card
                btnLeft="btnLeft"
                btnRight="btnRight"
                currentPosition="currentPosition"
                handleBtnClick={[Function]}
                id="0a985e0b-9383-43d7-bcf3-f8b8f597451e0"
                img="thumbnail"
                location="city"
                name="firstName0 lastName0"
              />,
            ],
            "className": "c-column__body",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0a985e0b-9383-43d7-bcf3-f8b8f597451e0",
              "nodeType": "class",
              "props": Object {
                "btnLeft": "btnLeft",
                "btnRight": "btnRight",
                "currentPosition": "currentPosition",
                "handleBtnClick": [Function],
                "id": "0a985e0b-9383-43d7-bcf3-f8b8f597451e0",
                "img": "thumbnail",
                "location": "city",
                "name": "firstName0 lastName0",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Column should render correctly with city filter 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Column
    btnLeft="btnLeft"
    btnRight="btnRight"
    currentPosition="currentPosition"
    data={
      Array [
        Object {
          "location": Object {
            "city": "city",
          },
          "login": Object {
            "uuid": "0a985e0b-9383-43d7-bcf3-f8b8f597451e0",
          },
          "name": Object {
            "first": "firstName0",
            "last": "lastName0",
          },
          "picture": Object {
            "thumbnail": "thumbnail",
          },
        },
      ]
    }
    filter="city"
    filterBy="city"
    handleBtnClick={[Function]}
    title="title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="c-column__header"
        >
          <h2
            className="c-column__title"
          >
            title
          </h2>
        </header>,
        <section
          className="c-column__body"
        >
          <Card
            btnLeft="btnLeft"
            btnRight="btnRight"
            currentPosition="currentPosition"
            handleBtnClick={[Function]}
            id="0a985e0b-9383-43d7-bcf3-f8b8f597451e0"
            img="thumbnail"
            location="city"
            name="firstName0 lastName0"
          />
        </section>,
      ],
      "className": "c-column",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h2
            className="c-column__title"
          >
            title
          </h2>,
          "className": "c-column__header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "title",
            "className": "c-column__title",
          },
          "ref": null,
          "rendered": "title",
          "type": "h2",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Card
              btnLeft="btnLeft"
              btnRight="btnRight"
              currentPosition="currentPosition"
              handleBtnClick={[Function]}
              id="0a985e0b-9383-43d7-bcf3-f8b8f597451e0"
              img="thumbnail"
              location="city"
              name="firstName0 lastName0"
            />,
          ],
          "className": "c-column__body",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0a985e0b-9383-43d7-bcf3-f8b8f597451e0",
            "nodeType": "class",
            "props": Object {
              "btnLeft": "btnLeft",
              "btnRight": "btnRight",
              "currentPosition": "currentPosition",
              "handleBtnClick": [Function],
              "id": "0a985e0b-9383-43d7-bcf3-f8b8f597451e0",
              "img": "thumbnail",
              "location": "city",
              "name": "firstName0 lastName0",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="c-column__header"
          >
            <h2
              className="c-column__title"
            >
              title
            </h2>
          </header>,
          <section
            className="c-column__body"
          >
            <Card
              btnLeft="btnLeft"
              btnRight="btnRight"
              currentPosition="currentPosition"
              handleBtnClick={[Function]}
              id="0a985e0b-9383-43d7-bcf3-f8b8f597451e0"
              img="thumbnail"
              location="city"
              name="firstName0 lastName0"
            />
          </section>,
        ],
        "className": "c-column",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h2
              className="c-column__title"
            >
              title
            </h2>,
            "className": "c-column__header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "title",
              "className": "c-column__title",
            },
            "ref": null,
            "rendered": "title",
            "type": "h2",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Card
                btnLeft="btnLeft"
                btnRight="btnRight"
                currentPosition="currentPosition"
                handleBtnClick={[Function]}
                id="0a985e0b-9383-43d7-bcf3-f8b8f597451e0"
                img="thumbnail"
                location="city"
                name="firstName0 lastName0"
              />,
            ],
            "className": "c-column__body",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0a985e0b-9383-43d7-bcf3-f8b8f597451e0",
              "nodeType": "class",
              "props": Object {
                "btnLeft": "btnLeft",
                "btnRight": "btnRight",
                "currentPosition": "currentPosition",
                "handleBtnClick": [Function],
                "id": "0a985e0b-9383-43d7-bcf3-f8b8f597451e0",
                "img": "thumbnail",
                "location": "city",
                "name": "firstName0 lastName0",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Column should render correctly with name filter 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Column
    btnLeft="btnLeft"
    btnRight="btnRight"
    currentPosition="currentPosition"
    data={
      Array [
        Object {
          "location": Object {
            "city": "city",
          },
          "login": Object {
            "uuid": "0a985e0b-9383-43d7-bcf3-f8b8f597451e0",
          },
          "name": Object {
            "first": "firstName0",
            "last": "lastName0",
          },
          "picture": Object {
            "thumbnail": "thumbnail",
          },
        },
      ]
    }
    filter="first"
    filterBy="name"
    handleBtnClick={[Function]}
    title="title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="c-column__header"
        >
          <h2
            className="c-column__title"
          >
            title
          </h2>
        </header>,
        <section
          className="c-column__body"
        >
          <Card
            btnLeft="btnLeft"
            btnRight="btnRight"
            currentPosition="currentPosition"
            handleBtnClick={[Function]}
            id="0a985e0b-9383-43d7-bcf3-f8b8f597451e0"
            img="thumbnail"
            location="city"
            name="firstName0 lastName0"
          />
        </section>,
      ],
      "className": "c-column",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h2
            className="c-column__title"
          >
            title
          </h2>,
          "className": "c-column__header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "title",
            "className": "c-column__title",
          },
          "ref": null,
          "rendered": "title",
          "type": "h2",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Card
              btnLeft="btnLeft"
              btnRight="btnRight"
              currentPosition="currentPosition"
              handleBtnClick={[Function]}
              id="0a985e0b-9383-43d7-bcf3-f8b8f597451e0"
              img="thumbnail"
              location="city"
              name="firstName0 lastName0"
            />,
          ],
          "className": "c-column__body",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0a985e0b-9383-43d7-bcf3-f8b8f597451e0",
            "nodeType": "class",
            "props": Object {
              "btnLeft": "btnLeft",
              "btnRight": "btnRight",
              "currentPosition": "currentPosition",
              "handleBtnClick": [Function],
              "id": "0a985e0b-9383-43d7-bcf3-f8b8f597451e0",
              "img": "thumbnail",
              "location": "city",
              "name": "firstName0 lastName0",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="c-column__header"
          >
            <h2
              className="c-column__title"
            >
              title
            </h2>
          </header>,
          <section
            className="c-column__body"
          >
            <Card
              btnLeft="btnLeft"
              btnRight="btnRight"
              currentPosition="currentPosition"
              handleBtnClick={[Function]}
              id="0a985e0b-9383-43d7-bcf3-f8b8f597451e0"
              img="thumbnail"
              location="city"
              name="firstName0 lastName0"
            />
          </section>,
        ],
        "className": "c-column",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h2
              className="c-column__title"
            >
              title
            </h2>,
            "className": "c-column__header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "title",
              "className": "c-column__title",
            },
            "ref": null,
            "rendered": "title",
            "type": "h2",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Card
                btnLeft="btnLeft"
                btnRight="btnRight"
                currentPosition="currentPosition"
                handleBtnClick={[Function]}
                id="0a985e0b-9383-43d7-bcf3-f8b8f597451e0"
                img="thumbnail"
                location="city"
                name="firstName0 lastName0"
              />,
            ],
            "className": "c-column__body",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0a985e0b-9383-43d7-bcf3-f8b8f597451e0",
              "nodeType": "class",
              "props": Object {
                "btnLeft": "btnLeft",
                "btnRight": "btnRight",
                "currentPosition": "currentPosition",
                "handleBtnClick": [Function],
                "id": "0a985e0b-9383-43d7-bcf3-f8b8f597451e0",
                "img": "thumbnail",
                "location": "city",
                "name": "firstName0 lastName0",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
